File: workflow/scripts/merge_datasets.py
# Send stdout and stderr to log file
import sys,os
import logging, traceback
logging.basicConfig(filename=snakemake.log[0],
                    level=logging.INFO,
                    format='%(asctime)s %(message)s',
                    datefmt='%Y-%m-%d %H:%M:%S',
                    )
def handle_exception(exc_type, exc_value, exc_traceback):
    if issubclass(exc_type, KeyboardInterrupt):
        sys.__excepthook__(exc_type, exc_value, exc_traceback)
        return

    logger.error(''.join(["Uncaught exception: ",
                         *traceback.format_exception(exc_type, exc_value, exc_traceback)
                         ])
                 )
# Install exception handler
sys.excepthook = handle_exception

# Import libraries

import pandas as pd

# Get variables

## Debug
IN = [
    "/hps/nobackup/birney/users/ian/MIKK_all/0.08/F0.csv",
    "/hps/nobackup/birney/users/ian/MIKK_all/0.08/F2.csv",
    "/hps/nobackup/birney/users/ian/MIKK_all/0.08/Kiyosu_CC.csv"
     ]

## True
IN = snakemake.input

File: workflow/scripts/merge_datasets.R
# Send log

log <- file(snakemake@log[[1]], open="wt")
sink(log)
sink(log, type = "message")

# Load libraries

library(tidyverse)

# Get variables

## Debug
IN = list(
    "/hps/nobackup/birney/users/ian/MIKK_all/0.08/F0.csv",
    "/hps/nobackup/birney/users/ian/MIKK_all/0.08/F2.csv",
    "/hps/nobackup/birney/users/ian/MIKK_all/0.08/Kiyosu_CC.csv"
)

## True
IN = snakemake@input
OUT = snakemake@output[[1]]

names(IN) = IN %>% 
  unlist() %>% 
  basename() %>% 
  stringr::str_remove(".csv")

# Read in files and combine

df = purrr::map_dfr(IN, readr::read_csv, .id = "dataset")

# Write to file

readr::write_csv(df, OUT)
